test/CompoundUtilTest.cc

```diff
@@ -0,0 +1,769 @@
+/*
+ * Copyright(c) 2019 Netflix, Inc.
+ * SPDX - License - Identifier: BSD - 2 - Clause - Patent
+ */
+
+/******************************************************************************
+ * @file CompoundUtilTest.cc
+ *
+ * @brief Unit test for util functions in wedge prediction:
+ * - av1_build_compound_diffwtd_mask_{, d16}avx2
+ * - aom_blend_a64_mask_avx2/aom_lowbd_blend_a64_d16_mask_avx2
+ * - aom_blend_a64_mask_sse4_1
+ * - aom_highbd_blend_a64_mask_sse4_1/aom_highbd_blend_a64_d16_mask_avx2
+ * - aom_blend_a64_hmask_sse4_1/aom_blend_a64_vmask_sse4_1
+ * - aom_highbd_blend_a64_hmask_sse4_1/aom_highbd_blend_a64_vmask_sse4_1
+ *
+ * @author Cidana-Wenyao
+ *
+ ******************************************************************************/
+#include "gtest/gtest.h"
+// workaround to eliminate the compiling warning on linux
+// The macro will conflict with definition in gtest.h
+#ifdef __USE_GNU
+#undef __USE_GNU  // defined in EbThreads.h
+#endif
+#ifdef _GNU_SOURCE
+#undef _GNU_SOURCE  // defined in EbThreads.h
+#endif
+#include "EbDefinitions.h"
+#include "EbUtility.h"
+#include "aom_dsp_rtcd.h"
+#include "random.h"
+#include "convolve.h"
+#include "util.h"
+
+using std::make_tuple;
+using svt_av1_test_tool::SVTRandom;
+
+#define MAKE_PARAM(func_type) std::tuple<func_type, func_type, const char *>
+
+namespace {
+template <typename SrcSample, typename DstSample, typename BlendFunc,
+          typename BlendTestParam>
+class CompBlendTest : public ::testing::TestWithParam<BlendTestParam> {
+  public:
+    CompBlendTest() {
+        tst_fn_name = "";
+        func_ref_ = nullptr;
+        func_tst_ = nullptr;
+        no_sub_ = false;
+    }
+
+    void SetUp() override {
+        ref_dst_ =
+            (DstSample *)eb_aom_memalign(32, MAX_SB_SQUARE * sizeof(DstSample));
+        tst_dst_ =
+            (DstSample *)eb_aom_memalign(32, MAX_SB_SQUARE * sizeof(DstSample));
+        src0_ =
+            (SrcSample *)eb_aom_memalign(32, MAX_SB_SQUARE * sizeof(SrcSample));
+        src1_ =
+            (SrcSample *)eb_aom_memalign(32, MAX_SB_SQUARE * sizeof(SrcSample));
+        mask_ = (uint8_t *)eb_aom_memalign(32, 4 * MAX_SB_SQUARE);
```

Comment:
It might be better if we define a variable like mask_size, and use it instead of 4 * MAX_SB_SQUARE.

Reply:
done